opendse-io/src/main/java/net/sf/opendse/io/TGFFReader.java

```diff
@@ -0,0 +1,424 @@
+package net.sf.opendse.io;
+
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import net.sf.opendse.model.Application;
+import net.sf.opendse.model.Architecture;
+import net.sf.opendse.model.Communication;
+import net.sf.opendse.model.Dependency;
+import net.sf.opendse.model.Link;
+import net.sf.opendse.model.Mapping;
+import net.sf.opendse.model.Mappings;
+import net.sf.opendse.model.Resource;
+import net.sf.opendse.model.Task;
+
+
+/**
+ * The {@code TgffReader} imports an {@code Application}, {@code Mappings} and a set of {@code Resource} types from a .tgff-file, 
+ * as generated by Task Graphs For Free (TGFF).
+ * 
+ * In particular, it serves to import benchmark task graphs, processor types and corresponding mapping possibilities from the 
+ * Embedded Systems Synthesis Benchmarks Suite (E3S).
+ * See <a href="http://ziyang.eecs.umich.edu/~dickrp/e3s/">http://ziyang.eecs.umich.edu/~dickrp/e3s/</a>
+ * 
+ * @author Valentina Richthammer 
+ */
+public class TGFFReader {
+	
+	private static final String HYPERPERIOD = "@HYPERPERIOD";

```

Comment:
could you please add a comment specifying when a static access to these private attributes is necessary?

Reply:
Again, there is not really a need for these attributes to be statically accessible; I implemented it this way, because I felt the attributes (which are tags in the TGFF-files) are rather associated with the class than with an instance of it, but I'm happy to change this as well.

Reply:
These are constants. Constants should be static final.

You could
* make the tgff keywords protected as they are only important if you want to extend the parser
* while the attribute names (PERIOD, ...) could be public static final as later, evaluators then could use them.